apiserver/facades/client/cloud/cloud.go

```diff
@@ -379,9 +414,125 @@ func (api *CloudAPI) Credential(args params.Entities) (params.CloudCredentialRes
 
 // AddCloud adds a new cloud, different from the one managed by the controller.
 func (api *CloudAPIV2) AddCloud(cloudArgs params.AddCloudArgs) error {
+	// TODO (anastasiamac 2018-02-16) This is needed as facade v2 has CAAS-specific calls
+	// but we need to bump facade version and make v3 available.
+	if !featureflag.Enabled(feature.CAAS) {
+		return errors.NotSupportedf("AddCloud call")
+	}
 	err := api.backend.AddCloud(common.CloudFromParams(cloudArgs.Name, cloudArgs.Cloud))
 	if err != nil {
 		return err
 	}
 	return nil
 }
+
+// CredentialContents returns the specified cloud credentials,
+// including the secrets if requested.
+// If no specific credential name/cloud was passed in, all credentials for this user
+// are returned.
+// Only credential owner can see its contents as well as what models use it.
+// Controller admin has no special superpowers here and is treated the same as all other users.
```

Comment:
Does the controller admin restriction really buy anything? What are we trying to hide? Secrets? They could just ssh into the controller and inspect the db.

Reply:
Of course, controller admin could but it's harder to inspect db, I guess?... :) Besides, if I only need to cater for credential owner access, I do not need to check for controller admin superpower at every turn.. :)

Reply:
controller admins are supposed to be able to access everything - we have several facades where we check for controller admin and treat that as the same as owner/admin. it would be surpising for that not be the case here too